### DISCUSSION TITLE
How to convert a stacked bar chart into a grouped bar chart?

### DISCUSSION THREAD

**Discussion by kevreth** [vega/vega-lite#7524] [Q&A] [answered]
This is coming out stacked. How can I convert it to grouped? Changing encoding.column.field to "type" produces a grouped bar chart but the years all appear twice.
```json
    {
      "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
      "data": {"values":
          [
            {
              "Year": 2002,
              "Rentals": 184,
              "VHS": 84
            },
            {
              "Year": 2003,
              "Rentals": 180,
              "VHS": 67
            }
          ]
      },
     "transform": [{
       "fold": ["Rentals", "VHS"],
       "as": ["type", "sales"]
      }],
      "mark": "bar",
      "encoding": {
        "x": {"field": "Year", "type": "ordinal"},
        "y": {"field": "sales", "type": "quantitative"},
        "color": {"field": "type", "type": "nominal"}
      }
    }
```

**Comment by domoritz** [comment, marked as answer]
[Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaoAJpwkAJzQAmAAy3rSgEpxZ6wQdQBGABwAWJQA1AAkAZTR-AF8FU3MrVDtbAGZnV3dPX1sgsLQANgB2SIBdaJBMCyRZCAAzNgsEI1BawWUjEBc3HQNs8KKlJE9DMoBPHDhFEAgdfRASvpBkCwBrNBAmSwnXKDZlGlkyNFAAD0OQapo4FtXMUfGlG7HVut3ZHRBS4dPzy9b0KcEZvdbqsAI4MSqYOhqGikd5KbaCOpfC5XdAPO4jR7oWRsBB7N6lBEMBDyTRnFG-EBmDZArEgZ74wTvSKRIA)

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "data": {"values":
      [
        {
          "Year": 2002,
          "Rentals": 184,
          "VHS": 84
        },
        {
          "Year": 2003,
          "Rentals": 180,
          "VHS": 67
        }
      ]
  },
 "transform": [{
   "fold": ["Rentals", "VHS"],
   "as": ["type", "sales"]
  }],
  "mark": "bar",
  "encoding": {
    "x": {"field": "type", "type": "ordinal"},
    "y": {"field": "sales", "type": "quantitative"},
    "color": {"field": "type", "type": "nominal"},
    "column": {
      "field": "Year", "type": "ordinal"
    }
  }
}
```

![image](https://user-images.githubusercontent.com/589034/121384372-0de00b80-c8fd-11eb-9390-c2f5a1857053.png)

Note: please use code fences, not `<code>`.

**Reply by kevreth** [reply]
Thanks. The final chart should resemble this, less the aesthetics. The chart above has the columns correct but looks like two charts together. This chart was made with Google Slides.
![Rentals and VHS](https://user-images.githubusercontent.com/47391465/121399732-eaf73b00-c980-11eb-9bdf-8bbde975afc3.png)

**Reply by jakevdp** [reply]
You can tweak the appearance if you like; see https://vega.github.io/vega-lite/examples/bar_grouped.html for an example.

**Reply by domoritz** [reply]
And better support for grouped bar charts will come when we implement the offset channel (#4703).

**Reply by kevreth** [reply]
> You can tweak the appearance if you like; see https://vega.github.io/vega-lite/examples/bar_grouped.html for an example.

Yes, I just don't know which tweaks to make.

**Reply by ezmiller** [reply]
Similarly to @kevreth I think, when I try this linked example, I end up with a stacked bar chart, rather than a grouped one.
